api/v1/views/states.py

```diff
@@ -1,28 +1,57 @@
 #!/usr/bin/python3
-"""State API endpoint"""
-from flask import abort, jsonify
+"""API endpoint"""
+from flask import abort, jsonify, request
 from api.v1.views import app_views
 from models import storage
+from models.state import State
 
 @app_views.route('/states', strict_slashes=False)
 def all_states():
     """Return list of all states"""
     all_states = storage.all("State")
     return jsonify([ obj.to_dict() for obj in all_states.values() ])
 
-@app_views.route('/states/<id>', strict_slashes=False)
-def state_by_id(id):
-    """Return State object based off id else raise 404"""
-    state = storage.get("State", id)
-    return jsonify(state.to_dict()) if state else abort(404)
-
-@app_views.route('/states/<id>', strict_slashes=False, methods=['DELETE'])
-def delete_state(id):
-    """Return State object based off id else raise 404"""
-    state = storage.get("State", id)
-    if state:
-        storage.delete()
-        storage.save()
-        return jsonify({}), 200
-    else:
+@app_views.route('/states', strict_slashes=False, methods=['POST'])
+def add_state():
+    """Add state to states"""
+    data = request.get_json()
+    name = data.get('name', None)
+    if not data:
+        return jsonify({'Error': "Not a JSON"}), 400
+    if not name:
+        return jsonify({'Error': "Missing name"}), 400
+
+    # this State already exists. Just update State with new data
+    for state in storage.all("State").values():
+        if state.name == name:
+            setattr(state, "name", name)
+            state.save()
+            return jsonify(state.to_dict()), 200
+
+    state = State(**data)
+    state.save()
+    return jsonify(state.to_dict()), 201
+
+@app_views.route('/states/<state_id>', strict_slashes=False, methods=['GET', 'PUT', 'DELETE'])
+def manipulate_state(state_id):
```

Comment:
Unless there's some reason while all of these actions on the state objects have to be in a single function, I think it would be clearer to have the different methods split out into different functions to keep the logic separate. Probably fine either way though.

Reply:
yeah, i did that at first but i saw my self repeating statements. thus combined it.